pkg/model/run.go

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2019 One Concern
+ *
+ */
+
+package model
+
+import (
+	"time"
+)
+
+const (
+	Version = 1
+)
+
+type Category struct {
+	Name        string      `json:"name" yaml:"name"`               // Unique name for the category
+	Message     string      `json:"message" yaml:"message"`         // Message describing the category.
+	Contributor Contributor `json:"contributor" yaml:"contributor"` // Credentials for who created the category.
+	Version     int         `json:"version" yaml:"version"`         // Version of the run Schema.
+}
+
+type Run struct {
+	Name           string      `json:"name" yaml:"name"`                     // Name for the run
+	InstanceID     string      `json:"instanceid" yaml:"instanceid"`         // External ID used for a run. Not guaranteed to be unique.
+	RunID          string      `json:"id" yaml:"id"`                         // Internally generated Unique id for an instance of a run. Each time the same Type+Name is run it results in a new ID.
+	Timestamp      time.Time   `json:"timestamp" yaml:"timestamp"`           // Timestamp for when the run was created.
+	Contributor    Contributor `json:"contributor" yaml:"contributor"`       // Contributor who created the run.
+	RunFile        string      `json:"runfile" yaml:"runfile"`               // Raw file describing the run. Example: A yaml file.
+	Version        int         `json:"version" yaml:"version"`               // Version of the run Schema.
+	PipelineEngine string      `json:"pipelineengine" yaml:"pipelineengine"` // The entity that executes the run.
+}
+
+type RunStatus struct {
+	Successful bool   `json:"successful" yaml:"successful"` // Successful run set to true if yes.
+	Log        string `json:"log" yaml:"log"`               // Log for the final status, can be the entire status as reported by the pipeline engine.
+}
+
+type FS struct {
+	Context    string `json:"context" yaml:"context"`       // Context for the dataset
+	Repo       string `json:"repo" yaml:"repo"`             // Repo for the dataset
+	BundleID   string `json:"bundleid" yaml:"bundleid"`     // BundleID to fetch
+	Label      string `json:"label" yaml:"label"`           // Label to fetch
+	Fuse       bool   `json:"fuse" yaml:"fuse"`             // Fuse mount for the dataset
+	Stream     bool   `json:"stream" yaml:"stream"`         // Stream the data in
+	VolumeName string `json:"volumename" yaml:"volumename"` // Volume name
+}
+
+type Container struct {
+	Image   string `json:"image" yaml:"image"`     // Image name
+	ImageID string `json:"imageid" yaml:"imageid"` // Image ID
+}
+
+type DataSetIn struct {
+	Timestamp   time.Time `json:"timestamp" yaml:"timestamp"`     // Timestamp for the dataset in
+	Stage       string    `json:"stage" yaml:"stage"`             // Stage in which the dataset is used
```

Comment:
Would the stage be the name of the individual step in the pipeline that the data is being used for?

Reply:
Such as "DelftModelCleanup" or something like that?

Reply:
Yes. I will add more documentation.